lib/runner/request-helpers-presend.js

```diff
@@ -150,20 +150,19 @@ module.exports = [
         url = request.url && request.url.toString();
 
         async.waterfall([
-            // try resolving through system proxy
+            // try resolving custom proxies before fall-backing to system proxy
             function (cb) {
-                _.isFunction(run.options.systemProxy) ? run.options.systemProxy(url, cb) : cb(null, undefined);
+                if (_.isFunction(_.get(proxies, 'resolve'))) {
+                    return cb(null, proxies.resolve(url));
+                }
+                return cb(null, undefined);
```

Comment:
this was already there (Deleted Line 155 )
do you want me to change it to `return cb()`?